app/v2/models/MovementType.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.models
+
+import play.api.libs.json.JsError
+import play.api.libs.json.JsString
+import play.api.libs.json.JsSuccess
+import play.api.libs.json.Reads
+import play.api.libs.json.Writes
+
+sealed abstract class MovementType(val movementType: String)
+
+object MovementType {
```

Comment:
can we not use movementType string available in MessageType trait? not sure if we need to create this new class if we have something available with messageType

Reply:
Quite the opposite really, the `MessageType` should really contain a `MovementType` as the movement type has become used in more than one place -- so I'll be updating it that way.

There is the problem where _that_ is a plural, but that's down to REST conventions and is correct in that instance, so the movement type will have that pluralised version as an additional field (but you'll still get the singular version).